lib/system/repr.nim

```diff
@@ -16,27 +16,27 @@ proc reprInt(x: int64): string {.compilerproc.} = return $x
 proc reprFloat(x: float): string {.compilerproc.} = return $x
 
 proc reprPointer(x: pointer): string {.compilerproc.} =
-  var buf: array[0..59, char]
-  discard c_sprintf(buf, "%p", x)
```

Comment:
I had problems with one of the methods, that they were executed on the VM, and I got the error in ``newString`` that casting is not allowed on the VM. I fixed ``newString`` to work on the VM, but I didn't like the idea to run a loop over characters on the VM, so I tried to replace the instances where this happened to work on strings directly rather than creating an array first and then convert it.